### ISSUE TITLE
Nvda not responding when settings dialog left open during exit

### ISSUE THREAD

**Issue by feerrenrut** [nvaccess/nvda#6172]
When leaving a settings dialog open (eg the Symbol list dialog) and exiting nvda there is a chance of a crash.

**Repro**
1. Start NVDA
2. open a settings dialog (nvda+n, p, p)
3. change focus to windows explorer 
4. exit nvda (nvda+q)

**Impact**
- The exit sound will not play.
- NVDA stops all output. 
- When the settings dialog is manually closed, a windows "application has stopped responding" dialog is shown.
- The log indicates that nvda is still processing keyboard input. In the attached log I press several keys to try and prompt speech from nvda after triggering the exit, since it seems to still be running.
- No crash dump seems to be created.

**Log**
[nvda-error.log.txt](https://github.com/nvaccess/nvda/files/360669/nvda-error.log.txt)

**Comment by feerrenrut** [comment]
I had focus highlight addon installed. After removing this nvda exits cleanly.

**Comment by bhavyashah** [comment]
@feerrenrut Does this mean that the reported crash is a result of a Focus Highlight add-on issue? In that case, I suppose we can safely close this ticket and report this to the add-ons community.

**Comment by Brian1Gaff** [comment]
I have noticed several occurrences of this in Windows 7. Yes I do have focus 
highlight installed, but even if an nvda  menu or settings sheet is open it 
can slow down windows  shutting down. Google drive has similar issues.
 It seems eventually to either manage to close or it gives you the option to 
force it, depending on the software.

Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.

**Comment by feerrenrut** [comment]
Yes, I believe this was a result of the focus highlight add-on. I have no tried this again with the latest focus-highlight addon: `focusHighlight-5.0-dev-170416 ` (which is at the pre-release stage). The crash did not occur. I believe I had `focusHighlight-3.0.nvda-addon` installed at the time. I tested this again with focus highlight 3 and cant reproduce the issue with nvda version `Version: master-14328,c6379eb3`. Closing this issue.
